src/busterbunny.js

```diff
@@ -70,14 +70,25 @@ module.exports = (function() {
         };
 
         self.raiseEvents = function (eventId, event, options, afterRaised) {
-            _stats.queuedEventsToRaise++;
-            if (options instanceof Function) {
+            if(options instanceof Function && afterRaised)
+                throw new Error('the third argument must be an object when a fourth argument is provided.');
+
+            if (options instanceof Function && !afterRaised) {
                 afterRaised = options;
                 options = null;
             }
 
-            var eventData = self.encoder.encodeEvent(event);
-            publishOrQueue(config.exchange, eventId, eventData, options, afterRaised);
+            if(!(afterRaised instanceof Function))
+                throw new Error('the argument provided for callback is not a Function');
+
+            if(typeof eventId !== 'string') {
+                afterRaised(new Error('eventId is required and must be a string'));
```

Comment:
Good point.  This choice was predicated that for any error returned by the framework (assuming a callback is provided) the callback should be called with the error.  The steps ahead of this ensure that we can at least call back with an error.  We should seek @binaryalchemist's opinion.